stingray/fourier.py

```diff
@@ -0,0 +1,1286 @@
+
+    for s, e, idx0, idx1 in fun(times, gti, segment_size):
+        if idx1 - idx0 < 2:
+            yield None
+            continue
+        if counts is None:
+            event_times = times[idx0:idx1]
+            # counts, _ = np.histogram(event_times - s, bins=bins)
+            # astype here serves to avoid integer rounding issues in Windows,
+            # where long is a 32-bit integer.
+            cts = histogram((event_times - s).astype(float), bins=N,
+                            range=[0, segment_size]).astype(float)
+            cts = np.array(cts)
+        else:
+            cts = counts[idx0:idx1].astype(float)
+            if errors is not None:
+                cts = cts, errors[idx0:idx1]
+
+        yield cts
+
+
+def avg_pds_from_iterable(flux_iterable, dt, norm="abs", use_common_mean=True, silent=False):
+    """Calculate the average periodogram from an iterable of light curves
+
+    Parameters
+    ----------
+    flux_iterable : `iterable` of `np.array`s or of tuples (`np.array`, `np.array`)
+        Iterable providing either equal segments of light curve, or of light curve
+        and errors. They must all be of the same length.
+    dt : float
+        Time resolution of the light curves used to produce periodograms
+
+    Other Parameters
+    ----------------
+    norm : str, default "abs"
+        The normalization of the periodogram. "abs" is absolute rms, "frac" is
+        fractional rms, "leahy" is Leahy+83 normalization, and "none" is the
+        unnormalized periodogram
+    use_common_mean : bool, default True
+        The mean of the light curve can be estimated in each interval, or on
+        the full light curve. This gives different results (Alston+2013).
+        Here we assume the mean is calculated on the full light curve, but
+        the user can set ``use_common_mean`` to False to calculate it on a
+        per-segment basis.
+    silent : bool, default False
+        Silence the progress bars
+
+    Returns
+    -------
+    freq : `np.array`
+        The periodogram frequencies
+    pds : `np.array`
+        The normalized periodogram powers
+    N : int
+        the number of bins in the light curves used in each segment
+    M : int
+        the number of averaged periodograms
+    mean : float
+        the mean counts per bin
+    """
+    local_show_progress = show_progress
+    if silent:
+
+        def local_show_progress(a):
+            return a
+
+    # Initialize stuff
+    cross = None
+    M = 0
+
+    sum_of_photons = 0
+    common_variance = None
+    for flux in local_show_progress(flux_iterable):
+        if flux is None:
+            continue
+
+        # If the iterable returns the uncertainty, use it to calculate the variance.
+        variance = None
+        if isinstance(flux, tuple):
+            flux, err = flux
+            variance = np.mean(err) ** 2
+
+        # Calculate the FFT
+        N = flux.size
+        ft = fft(flux)
+
+        nph = flux.sum()
```

Comment:
So in other functions, this parameter is `Nph`. Should we make it lower-case there as well or capitalize here? I feel like we now have a large enough code base that being as consistent as possible just makes things easier on ourselves if we can track equivalent quantities across functions/classes/modules.

Reply:
I changed these and all the others to ``n_ph``